python/ml4ir/base/model/calibrations/temperature_scaling.py

```diff
@@ -0,0 +1,247 @@
+
+    start = tf.Variable(initial_value=[temperature_init], shape=(1), dtype=tf.float32, name="temp",
+                        trainable=True)
+
+    def nll_loss_fn(temperature):
+        """
+        computes Negative Log Likelihood loss by applying temperature scaling.
+        Parameters:
+            temperature: TF.Tensor,  a  parameter of size=(1) to be trained.
+        """
+        logits_tensor = tf.constant(logits_nps, name='logits_tensor', dtype=tf.float32)
+        labels_tensor = tf.constant(labels_nps, name='labels_tensor', dtype=tf.int32)
+        logits_w_temp = tf.divide(logits_tensor, temperature)
+
+        return tf.reduce_mean(
+            tf.nn.sparse_softmax_cross_entropy_with_logits(labels=labels_tensor,
+                                                           logits=logits_w_temp))
+
+    @tf.function
+    def nll_with_lbfgs():
+        return tfp.optimizer.lbfgs_minimize(make_val_and_grad_fn(nll_loss_fn),
+                                            initial_position=start)
+
+    # evaluation on validation set before temperature scaling
+    logits_nps, labels_nps = get_logits_labels(intermediate_model, dataset.validation)
+    org_acc_op, org_nll_loss_op, _ = eval_relevance_model(scorer, logits_nps,
+                                                          labels_nps)
+
+    # perform temperature scaling
+    results = run(nll_with_lbfgs, logger)
+
+    # evaluation on validation set with temperature scaling
+    temper = tf.constant(results.position)
+    acc_op, nll_loss_op, _ = eval_relevance_model(scorer, logits_nps, labels_nps,
+                                                  temperature=temper)
+
+    logger.info("=" * 50)
+    logger.info(f'temperature value : {results.position}')
+    logger.info('Evaluating on the validation dataset ')
+    logger.info(f'original loss: {org_nll_loss_op}, accuracy: {org_acc_op}, \n'
+          f'temperature scaling loss: {results.objective_value}, accuracy: {acc_op}\n')
+
+    logger.info("="*50)
+    logger.info("Evaluating on the test dataset")
+
+    logits_nps_test, labels_nps_test = get_logits_labels(intermediate_model, dataset.test)
+
+    # evaluation on test set before temperature scaling
+    acc_test_ts, _, org_softmaxes = eval_relevance_model(scorer, logits_nps_test,
+                                          labels_nps_test)
+
+    # evaluation on test set with temperature scaling
+    acc_test_org, _, ts_softmaxes = eval_relevance_model(scorer,
+                                                         logits_nps_test,
+                                           labels_nps_test, temperature=temper)
+
+    # write the full vector in the csv not ...
+    np.set_printoptions(
+        formatter={'all': lambda x: str(x.decode('utf-8')) if isinstance(x, bytes) else str(x)},
+        linewidth=sys.maxsize,
+        threshold=sys.maxsize)
+    # avoiding .tolist() that is not memory efficient
+    data_dic = {'org_scores': [x for x in org_softmaxes.numpy()],
```

Comment:
why do we call them 'org_scores' ? Same, below for 'ts_scores'.

Reply:
`org_score`: original score (without temperature scaling)
`ts_score`: scores with temperature scoring

Reply:
```suggestion
    data_dic = {'original_scores': [x for x in org_softmaxes.numpy()],
```

Reply:
As a general rule, I would use full names everywhere unless I have a good reason not to.